Caboodle/Battery/Battery.android.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Android;
+using Android.Content;
+using Android.OS;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Battery
+    {
+        static bool hasBatteryStatsPermission;
+        static void ValidateBatteryStatsPermission()
+        {
+            if (hasBatteryStatsPermission)
+                return;
+
+            var permission = Manifest.Permission.BatteryStats;
+            if (!Platform.HasPermissionInManifest(permission))
+                throw new PermissionException(permission);
+
+            hasBatteryStatsPermission = true;
+        }
+        public static double ChargeLevel
+        {
+            get
+            {
+                ValidateBatteryStatsPermission();
+
+                using (var filter = new IntentFilter(Intent.ActionBatteryChanged))
+                {
+                    using (var battery = Platform.CurrentContext.RegisterReceiver(null, filter))
```

Comment:
You don't need multiple indents:

```csharp
using (var filter = new IntentFilter(Intent.ActionBatteryChanged)) 
using (var battery = Platform.CurrentContext.RegisterReceiver(null, filter))
{
}
```

Reply:
IDK that looks confusing to me. I like it nested just makes it cleaner to read in my honest opinion.